platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/EventRecoveryWorkflow.java

```diff
@@ -378,8 +377,8 @@ private static ReservedSignedState handleNextRound(
         final Instant currentRoundTimestamp = getRoundTimestamp(round);
         previousState.get().getState().throwIfImmutable();
         final MerkleRoot newState = previousState.get().getState().copy();
-        final DetailedConsensusEvent lastEvent = (DetailedConsensusEvent) getLastEvent(round);
-        new StatefulEventHasher().hashEvent(lastEvent.getPlatformEvent());
+        final PlatformEvent lastEvent = (PlatformEvent) getLastEvent(round);
+        new StatefulEventHasher().hashEvent(lastEvent);
```

Comment:
Is this where the event's hash is set?  I'm not familiar with this code enough to determine if this is the right solution.  I think @cody-littley  or @litt3 should weigh in.

Reply:
This hashing bit seems alright to me (since it's where the hash was being set in this workflow before, too)

As to the PR in general, it's not clear to me why `PlatformEvent` is being modified.

Reply:
I've talked with Lazar and we changed the Event Iterator